arbos/block_processor.go

```diff
@@ -313,22 +291,6 @@ func ProduceBlockAdvanced(
 				return nil, nil, err
 			}
 
-			if chainConfig.ArbitrumChainParams.EnableEspresso {
```

Comment:
okay makes sense! I will re-add it - thank you

Reply:
For my own understanding, where can I read more about builder guaranteed transactions? @sveitser

Reply:
Actually after thinking more about it, I dont think code is needed because espressoHeader will always be nil. The only time the espressoHeader was not nil was when inside `ProduceBlock` we had this block of code:

```
var espressoHeader *espressoTypes.Header
	if chainConfig.ArbitrumChainParams.EnableEspresso {
		if IsEspressoMsg(message) && !IsEspressoSovereignMsg(message) {
			// creating a block with espresso message
			_, jst, err := ParseEspressoMsg(message)
			if err != nil {
				return nil, nil, err
			}
			espressoHeader = jst.Header
		}
	}
```

which is now redundant. Let me know your thoughts on the same!

Reply:
There is a page here: https://www.notion.so/espressosys/Builder-guaranteed-transactions-formerly-Atomic-Blocks-fb49b6ec83f5458c821804f91ebf3de2

Reply:
I have created an issue for the same [here](https://github.com/EspressoSystems/nitro-espresso-integration/issues/220)